pkg/cloud/gcp/actuators/machine/reconciler.go

```diff
@@ -515,21 +532,27 @@ func (r *Reconciler) delete() error {
 		return err
 	}
 
-	// Remove instance from instance group, if necessary
-	if r.machineScope.machine.Labels[openshiftMachineRoleLabel] == masterMachineRole {
-		if err := r.unregisterInstanceFromControlPlaneInstanceGroup(); err != nil {
-			return fmt.Errorf("%s: failed to unregister instance from instance group: %v", r.machine.Name, err)
-		}
-	}
-
+	// Make sure that the machine exists.
+	// Also check that we have a machine with valid configuration.
 	exists, err := r.exists()
-	if err != nil {
+	if exists && isInvalidMachineConfigurationError(err) {
+		return fmt.Errorf("the machine %s has invalid configuration, but already exists, make the configuration of the machine valid for the deletion to be successful", r.machine.Name)
+	}
+	if err != nil && !isInvalidMachineConfigurationError(err) {
```

Comment:
If the config is invalid and there's a providerID, we aren't going to proceed as we think there's already a machine present. We don't use `exists` because actually, we don't know if it does exist or not, I think this makes it more explicit.
```suggestion
	if isInvalidMachineConfigurationError(err) && machine.Spec.ProviderID != nil {
		return fmt.Errorf("the machine %s has invalid configuration, but already exists, make the configuration of the machine valid for the deletion to be successful", r.machine.Name)
	}
	if err != nil && !isInvalidMachineConfigurationError(err)  {
```